capstone-backend/app.js

```diff
@@ -137,56 +138,340 @@ app.post ('/profile/interests', async (req, res) => {
         }
       }
-      res.send({skills: skills, userInfo: currentUser, message: "interests saved!", typeStatus: "success", infoInterests: infoInterests})
+      if (uiLanguage){
+        const query = new Parse.Query(Language);
+        query.equalTo("User1", currentUser);
+        query.equalTo("name", uiLanguage.name);
+        const results = await query.find();
+        if (!results[0]){
+          language.set("name", uiLanguage.name);
+          let relations = language.relation("User1");
+          relations.add(currentUser);
+          await language.save()
+        }
+      }
+      res.send({skills: respSkills, company: respCompany, language: language,  userInfo: currentUser, message: "interests saved!", typeStatus: "success", infoInterests: infoInterests})
     } else {
-      res.send({skills: skills, userInfo: null, message: "cannot find a user currently", typeStatus: "danger", infoInterests: infoInterests})
+      res.send({skills: null , userInfo: null, message: "cannot find a user currently", typeStatus: "danger", infoInterests: infoInterests})
     }
   }
   catch(error){
     res.send( {message: error.message, infoInterests: infoInterests, typeStatus: "danger"})
   }
 })
 
+app.post('/profile/interests/remove', async (req, res) => {
+  const removeInfo = req.body
+  Parse.User.enableUnsafeCurrentUser();
+  const currentUser = Parse.User.current();
+  try {
+    if (currentUser){
+      if (removeInfo.skills){
+        removeInterest("Skills", "name", removeInfo.skills.name, currentUser)
+      }
+      res.send({message : "success", removeInfo: removeInfo, result: result[0]})
+    }
+    else {
+      res.send({message : "no user found", typeStatus: "danger"})
+    }
+  }
+  catch (error){
+    res.send({message: error.message, typeStatus: "danger"})
+  }
+})
+
+app.post('/matches', async (req, res) => {
+  Parse.User.enableUnsafeCurrentUser();
+  const params = req.body.params;
+  const currentUser = Parse.User.current()
+
+  try {
+    if (currentUser){
+      if (params.liked){
+        updateMatch(params, currentUser)
+      }
+      else {
+        getMatches(currentUser);
+      }
+      res.send({message: "success match", typeStatus: "success"})
+    }
+    else {
+      res.send({message: "no user found!", typeStatus: "danger"})
+    }
+  }
+  catch (error){
+    res.send({message: error.message, typeStatus: "danger"})
+  }
+})
+
+app.get('/matches', async (req, res) => {
+  Parse.User.enableUnsafeCurrentUser();
+  const currentUser = Parse.User.current()
+  const limit = req.query["limit"];
+  const offset = req.query["offset"];
+
+  try {
+    if (currentUser) {
+      let matchData = await getMatchData(limit, offset, currentUser);
+      res.send(matchData)
+    }
+    else {
+      res.send({message : "no user found!" , typeStatus: "danger"})
+    }
+  }
+  catch (error){
+    res.send({message: "cannot get match data", typeStatus: "danger"})
+  }
+})
+
+async function updateMatch(params, currentUser){
```

Comment:
All the functions from here are well defined. It would be better to move them to an external file and import them here. 
That would help with a particular problem: app.js keeps growing and growing and contains all of the business logic.